tests/php/Extension/FluentSiteTreeExtensionTest.php

```diff
@@ -334,140 +332,94 @@ function (FluentState $newState) use ($localeCode, $fixture, $expected) {
+     * @param bool $frontendContext
+     * @param bool $expected
+     * @return void
+     * @dataProvider localeInheritanceModeCasesProvider
+     */
+    public function testLocaleInheritanceMode(
+        string $cmsInheritanceMode,
+        string $frontendInheritanceMode,
+        bool $frontendContext,
+        bool $expected
+    ): void {
+        Page::config()
+            ->set('cms_localisation_required', $cmsInheritanceMode)
+            ->set('frontend_publish_required', $frontendInheritanceMode);
+
+        FluentState::singleton()->withState(function (FluentState $state) use ($frontendContext, $expected) {
+            $state
                 ->setLocale('de_DE')
                 ->setIsDomainMode(false)
-                ->setIsFrontend(false);
+                ->setIsFrontend($frontendContext);
 
-            $page = Page::get()->filter('URLSegment', 'home')->first();
+            $page = Page::get()->find('URLSegment', 'home');
 
-            $this->assertNotNull($page);
-        });
-    }
-
-    public function testHomeNotVisibleInCMSBothConfigExact()
-    {
-        Config::modify()->set(DataObject::class, 'cms_localisation_required', FluentExtension::INHERITANCE_MODE_EXACT);
-        Config::modify()->set(DataObject::class, 'frontend_publish_required', FluentExtension::INHERITANCE_MODE_EXACT);
+            if ($expected) {
+                $this->assertNotNull($page, 'We expect the page model to have content available');
 
-        FluentState::singleton()->withState(function (FluentState $newState) {
-            $newState
-                ->setLocale('de_DE')
-                ->setIsDomainMode(false)
-                ->setIsFrontend(false);
-
-            $page = Page::get()->filter('URLSegment', 'home')->first();
+                return;
+            }
 
-            $this->assertNull($page);
+            $this->assertNull($page, 'We expect the page model to not have any content available');
         });
     }
 
-    public function testHomeNotVisibleInCMSOneConfigExact()
+    public function localeInheritanceModeCasesProvider(): array
     {
-        Config::modify()->set(DataObject::class, 'cms_localisation_required', FluentExtension::INHERITANCE_MODE_EXACT);
-        Config::modify()->set(DataObject::class, 'frontend_publish_required', FluentExtension::INHERITANCE_MODE_ANY);
-
-        FluentState::singleton()->withState(function (FluentState $newState) {
-            $newState
-                ->setLocale('de_DE')
-                ->setIsDomainMode(false)
-                ->setIsFrontend(false);
-
-            $page = Page::get()->filter('URLSegment', 'home')->first();
-
-            $this->assertNull($page);
-        });
+        return [
+            'cms with any mode, frontend with any mode, frontend context' => [
+                FluentExtension::INHERITANCE_MODE_ANY,
+                FluentExtension::INHERITANCE_MODE_ANY,
+                true,
+                true,
```

Comment:
```suggestion
                'cmsInheritanceMode' => FluentExtension::INHERITANCE_MODE_ANY,
                'frontendInheritanceMode' => FluentExtension::INHERITANCE_MODE_ANY,
                'frontendContext' => true,
                'expected' => true,
```
This is needed for PHPUnit 11 which is used in CMS 6 - so doing this now means less work on merge up to the `8.0` branch.
It also makes it immediately clear what each argument is for, which is useful if one of these scenarios starts failing unexpectedly.

Please make this change to all of the scenarios in this provider.